src/app/api/stories/id/[id]/route.ts

```diff
@@ -0,0 +1,179 @@
+import prisma from "@/lib/prisma";
+import { type ContributionType, type Prisma, type Story } from "@prisma/client";
+import { getServerSession } from "next-auth";
+import { NextResponse, type NextRequest } from "next/server";
+import { patchStorySchema } from "../../schema";
+
+interface Params {
+  id: unknown;
+}
+
+export type GetStoryResult = Story & {
+  storyContributions: {
+    user: {
+      id: string;
+      firstName: string;
+      lastName: string;
+      bio: string;
+      avatarUrl: string | null;
+    };
+    contributionType: ContributionType;
+  }[];
+  storyContent: {
+    content: string;
+    footer: string | null;
+  }[];
+};
+
+export async function GET(req: NextRequest, { params }: { params: Params }) {
+  try {
+    const { searchParams } = new URL(req.url);
+
+    const { id } = params;
+
+    if (typeof id !== "string") {
+      return NextResponse.json({ error: "Bad Request" }, { status: 400 });
+    }
+
+    const includeContent = searchParams.get("include_content");
+
+    const result = await prisma.story.findUnique({
+      where: {
+        id,
+      },
+      include: {
+        storyContributions: {
+          select: {
+            contributionType: true,
+            user: {
+              select: {
+                id: true,
+                firstName: true,
+                lastName: true,
+                bio: true,
+                avatarUrl: true,
+              },
+            },
+          },
+        },
+        storyContent:
+          includeContent === "true"
+            ? {
+                take: 1,
+                orderBy: { createdAt: "desc" },
+                select: { content: true, footer: true },
+              }
+            : false,
+      },
+    });
+
+    if (result === null) {
+      return NextResponse.json({ error: "Not Found" }, { status: 404 });
+    }
+
+    return NextResponse.json(result);
+  } catch (err) {
+    console.error(err);
+    return NextResponse.json(
+      { error: "Internal Server Error" },
+      { status: 500 },
+    );
+  }
+}
+
+export async function PATCH(
+  request: NextRequest,
+  { params }: { params: Params },
+) {
+  try {
+    const session = await getServerSession();
+    const user = await prisma.user.findUnique({
+      where: { email: session?.user.email ?? "noemail" },
+    });
+
+    if (!user || !user.roles.includes("EDITOR")) {
```

Comment:
so I notice some places here we are checking specifically if a user is an editor but not if they're an author.  Do authors not get the ability to edit articles?  Or are all authors also editors?  This probably doesn't matter for this pull request, but I was uncertain about the distinction.